src/components/ToggleSwitch.js

```diff
@@ -93,73 +60,103 @@ const ToggleSwitch = React.createClass({
   _renderRightLabel (styles) {
     if (this.props.showLabels) {
       return (
-        <span className='right-label' onClick={this._handleRightLabelClick} style={[styles.label, this.props.labelStyle, this.state.activePosition === 'right' && styles.activeLabel || styles.inactiveLabel]}>{this.props.rightLabel}</span>
+        <span className='right-label' onClick={this._handleToggle} style={[styles.label, this.state.checked && styles.activeLabel || styles.inactiveLabel]}>{this.props.rightLabel}</span>
       );
     } else {
       return null;
     }
   },
 
-  _getChildren () {
-    // Let all the children of the toggle switch be aware of the activePosition
-    return React.Children.map(this.props.children, child => {
-      return React.cloneElement(child, { activePosition: this.state.activePosition });
-    });
+  _renderIcons (styles) {
+    if (!this.props.showIcons) {
+      return null;
+    }
+    const trueIcon = this.props.trueIcon || <Icon className='true-icon' style={Object.assign({}, styles.icon, styles.trueIcon)} type='check-skinny' />;
+    const falseIcon = this.props.falseIcon || <Icon className='false-icon' style={Object.assign({}, styles.icon, styles.falseIcon)} type='close-skinny' />;
+
+    return (
+      <span>
+        {trueIcon} {falseIcon}
+      </span>
+    );
   },
 
   render () {
-    const styles = {
-      activeLabel: {
-        color: this.props.activeColor
-      },
+    const styles = Object.assign({}, {
```

Comment:
We should move the styles object to it's own styles function like we do internally so we don't have to pass it around as an argument.